rust_src/src/remacs_sys.rs

```diff
@@ -1278,9 +1279,1303 @@ extern "C" {
     pub fn fset_selected_window(frame: *mut Lisp_Frame, window: LispObject);
 }
 
+#[repr(C)]
+#[derive(Debug, Copy, Clone)]
+pub struct kboard {
+    _unused: [u8; 0],
+}
+
+#[repr(u32)]
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash)]
+pub enum output_method {
+    output_initial = 0,
+    output_termcap = 1,
+    output_x_window = 2,
+    output_msdos_raw = 3,
+    output_w32 = 4,
+    output_ns = 5,
+}
+
+#[repr(C)]
+#[derive(Debug, Copy)]
+pub struct image_cache {
+    pub buckets: *mut *mut image,
+    pub images: *mut *mut image,
+    pub size: isize,
+    pub used: isize,
+    pub refcount: isize,
+}
+#[test]
+fn bindgen_test_layout_image_cache() {
```

Comment:
bindgen?

Reply:
Yep. I used bindgen to generate them, since typing them all in manually would be annoying and quite error-prone.

Obviously at some point we'll actually use bindgen to generate these bindings during the build rather than checking them in, but this will do for now.

Reply:
ah. Thanks.